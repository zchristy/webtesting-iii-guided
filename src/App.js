import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="app">
        <h2>{this.props.flag}</h2>
      </div>
    );
  }
}

export default App;
