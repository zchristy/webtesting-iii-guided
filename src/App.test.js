import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer'; // 1: install this npm module as a dev dependency

import App from './App';

describe('<App />', () => {
  it('runs tests', () => {
    expect(true).toBe(true);
  });

  // 2. write this test
  // it('matches snapshot', () => {
  //   const tree = renderer.create(<App flag={'no'} />); // generates a DOM tree
  //   // snapshots are a JSON representation of the DOM tree
  //   expect(tree.toJSON()).toMatchSnapshot();
  // });

  it('matches snapshot', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
