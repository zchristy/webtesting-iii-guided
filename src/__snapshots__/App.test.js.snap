// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div>
  <div
    class="app"
  >
    <h2 />
  </div>
</div>
`;
